uxarray/grid/grid.py

```diff
@@ -96,211 +93,77 @@ def __init__(self, input_obj, **kwargs):
         # initialize cached data structures (nearest neighbor operations)
         self._ball_tree = None
 
-        # unpack kwargs with default values set to None
-        kwargs_list = [
-            'gridspec', 'vertices', 'islatlon', 'isconcave', 'source_grid',
-            'use_dual'
-        ]
-        for key in kwargs_list:
-            setattr(self, key, kwargs.get(key, None))
-
-        # check if initializing from verts:
-        if isinstance(input_obj, (list, tuple, np.ndarray)):
-            input_obj = np.asarray(input_obj)
-            self.mesh_type = "From vertices"
-            # grid with multiple faces
-            if input_obj.ndim == 3:
-                self.__from_vert__(input_obj)
-                self.source_grid = "From vertices"
-            # grid with a single face
-            elif input_obj.ndim == 2:
-                input_obj = np.array([input_obj])
-                self.__from_vert__(input_obj)
-                self.source_grid = "From vertices"
-            else:
-                raise RuntimeError(
-                    f"Invalid Input Dimension: {input_obj.ndim}. Expected dimension should be "
-                    f"3: [nMesh2_face, nMesh2_node, Two/Three] or 2 when only "
-                    f"one face is passed in.")
-
-        # check if initializing from string
-        # TODO: re-add gridspec initialization when implemented
-        elif isinstance(input_obj, xr.Dataset):
-            self.mesh_type = _parse_grid_type(input_obj)
-            self.__from_ds__(dataset=input_obj)
-        else:
-            raise RuntimeError("Dataset is not a valid input type.")
+    @classmethod
+    def from_dataset(cls,
+                     dataset: xr.Dataset,
+                     use_dual: Optional[bool] = False):
+        """Constructs a ``Grid`` object from an ``xarray.Dataset``.
 
-        # {"Standardized Name" : "Original Name"}
-        self._inverse_grid_var_names = {
-            v: k for k, v in self.grid_var_names.items()
-        }
+        Parameters
+        ----------
+        dataset : xr.Dataset
+            ``xarray.Dataset`` containing unstructured grid coordinates and connectivity variables
+        use_dual : bool, default=False
+            When reading in MPAS formatted datasets, indicates whether to use the Dual Mesh
+        """
+        if not isinstance(dataset, xr.Dataset):
+            raise ValueError("Input must be an xarray.Dataset")
+
+        # determine grid/mesh specification
+        grid_spec = _parse_grid_type(dataset)
+
+        if grid_spec == "Exodus":
+            grid_ds, ugrid_mapping = _read_exodus(dataset)
```

Comment:
Agreed, it might be a better choice to have a `ugrid_dim_mapping` that maps the dimensions (nodes, faces, edges) to the respective name in the source dataset if they differ from the ugrid conventions.

The goal of this was to use it within the Dataset and DataArray to rename any node, edge, face dimension to their UGRID equivalent.

Reply:
> I mean, we shouldn't attempt on tracking the source names for the formats other than UGRID. The code should look like:

The reason for tracking the `source_dims_dict` across all grid formats is for the case where we end back up in our `UxDataset` and `UxDataArray`, the dimensions that we use within our `Grid` will differ from the dimensions used in those datasets, if they originally come in a different format. 

This is what was discussed in #363

Reply:
Oh, I just got the actual concern in #363 and so what you are doing here now; thanks a lot for clarifying that! 

So, okay, we are trying to use standardized dim names consistently, i.e. not only in `Grid` but also in `UxDataset` and `UxDataArray`. I have a few questions and comments then:

1. Why only dim names. Shouldn't we also use var names etc. in the standardized way in our `UxDataset` and `UxDataArray` as well?
2. This is good for satisfying #363, but we will need to handle a further thing with our standardized names. Assume the following scenario:
    - Read a UGRID dataset with custom names into `uxds`, so we will be changing those names to our standardized names.
    - Call `uxds.uxgrid.encode_as("ugrid")` that will output the grid in UGRID format as type xarray.Dataset
       - Problem: It will have our standardized names showing up; however, the user expects to see the original ugrid dataset's custom names because the flow was simply: read in ugrid, encode as ugrid; why variable/dim name changes?
3. If you agree with above cases needing to be addressed, then should we separate this dim/var name mapping things from this PR as this is a big task?

Reply:
> Why only dim names. Shouldn't we also use var names etc. in the standardized way in our UxDataset and UxDataArray as well?

Since the purpose of our UxDataset and UxDataArray is to store _data variables_, there is "standardized" name for those (other than standardizing the dimensions, which is already done"

> remap_dims = True makes sense, but I am not sure we should even consider providing such an optional argument that could take False value. I can't see use cases with remap_dims = False, so should we handle it like always remapping the dims?

I can remove it and make it always remap if we prefer 

> the flow was simply: read in ugrid, encode as ugrid; why variable/dim name changes?

I don't believe this should be something we need to worry about, especially for the following case
1. Read in UGRID
2. Do stuff (i.e. compute `Mesh2_face_edges`, compute face area)
3. Encode as UGRID

If we chose to encode using the original variable and dimension names, there is a high likelyhood that the encoded dataset will have two types of variables/dims (original mixed with new ones from the computed variables).

for the sake of standardization, I think its fair to assume that our `encode_as` functions will follow our definition of an unstructured grid, including the variable and dimension names.

On the topic of our "definition", we should look into better documenting our definition of an unstructured grid, because even though we follow the UGRID conventions, we have been slowly adding our own variables, such as the way we represent Cartesian Coordinates. 

Once this is merged, I'll handle that in #408 

> If you agree with above cases needing to be addressed, then should we separate this dim/var name mapping things from this PR as this is a big task?

This might be worth revisiting, however for the sake of this PR the removal of `grid_var_names` was to deprecate it's existing inconsistent internal usage. We can consider reintroducing a variant down the line if needed.

Reply:
> Since the purpose of our UxDataset and UxDataArray is to store _data variables_, there isn't "standardized" name for those (other than standardizing the dimensions, which is already done".

I was not clear enough about my use of the word "vars". I meant the cases where UxDataset has also the grid topology variables (e.g. latCell, lonCell, cellsOnEdges, etc.) in it, in which it feels a bit awkward to see the dataset with those source grid format's original names while we have the `uxds.uxgrid` have our standardized names (I know we are supposed to always use grid topology info through `.uxgrid`, but it may still get confusing in the user's end, for instance when hovering over the `uxds` in an IDE). 
 
> > remap_dims = True makes sense, but I am not sure we should even consider providing such an optional argument that could take False value. I can't see use cases with remap_dims = False, so should we handle it like always remapping the dims?
> 
> I can remove it and make it always remap if we prefer

If that also makes sense to you, I think that's better.

> > the flow was simply: read in ugrid, encode as ugrid; why variable/dim name changes?
> 
> I don't believe this should be something we need to worry about, especially for the following case
> 
> 1. Read in UGRID
> 2. Do stuff (i.e. compute `Mesh2_face_edges`, compute face area)
> 3. Encode as UGRID
> 
> If we chose to encode using the original variable and dimension names, there is a high likelyhood that the encoded dataset will have two types of variables/dims (original mixed with new ones from the computed variables).
> 
> for the sake of standardization, I think its fair to assume that our `encode_as` functions will follow our definition of an unstructured grid, including the variable and dimension names.
> 
> On the topic of our "definition", we should look into better documenting our definition of an unstructured grid, because even though we follow the UGRID conventions, we have been slowly adding our own variables, such as the way we represent Cartesian Coordinates.
> 
> Once this is merged, I'll handle that in #408
 
Actually, your points make much sense to me. However, the following quote was from #363, so I think we should still take into consideration there might be some user expectations about this:

> "Converting (or standardizing) names would certainly make the matter simple inside of uxarray, though it means that we are losing a bit of the original UGRID information.
Keeping the original information in post-processed outputs may matter in some cases. (I certainly need to keep it, at least for now.) Uxarray does not (or does not need to) have writing IO functions implemented yet, as far as I know, and it is probably not a concern for now."

Reply:
@erogluorhan 

> I meant the cases where UxDataset has also the grid topology variables (e.g. latCell, lonCell, cellsOnEdges, etc.)

It would makes sense to be able to "filter" these out of the UXdataset to prevent duplicates between the Grid and Dataset, so this is worth considering in a future PR

> Actually, your points make much sense to me. However, the following quote was from https://github.com/UXARRAY/uxarray/issues/363, so I think we should still take into consideration there might be some user expectations about this:

Got it! I will give it some thought and raise an issue best addressing how to provide a solution to this.

Reply:
Okay, let me create issues for both of these topics; we can follow up on them in the future

Reply:
Okay, created #449 and #450 for these